### DISCUSSION TITLE
SD_mp3_FIR filter_i2s

### DISCUSSION THREAD

**Discussion by twillging** [pschatzmann/arduino-audio-tools#1172] [Q&A] [answered]
Hi all, I'm wondering if anyone has been able to implement a simple low-pass filter on decoded mp3 audio and output it on the i2s. I have working code for reading and decoding mp3 from the SD and outputting on the i2s--and I've even been able to send it through the volume stream before outputting to i2s. 

I've been constantly failing to implement the filter, though. Could anyone provide the appropriate steps here? Ideally, I'd like to decode mp3 from the SD card, run it through a LPF, and then output the filtered audio to the i2s. Even more ideal, the user could turn on and off the filter (could probably just switch the filter coeffs to zero?). 

Thanks in advance for any help on this.

**Comment by pschatzmann** [comment, marked as answer]
I don't quite understand where your problem is. 
Did you get any error messages ?
The chain e.g. would be File - copy -> EncodedAudioStream -> FilteredStream -> I2SStream
and the [Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/Filters) explains how to set up and use filters
You can dynamically set the output for the EncodedAudioStream either to the FilteredStream or the I2SStream object to activate, deactivate the filter.

**Reply by twillging** [reply]
I think I may have implemented it correctly before but was unable to notice the effect of the filter. I retried just now (implemented the volume stream and then switched it out for a filtered stream) and checked that the filter was working by changing all coeffs to zero, and it muted the output. So I'd say it's working now. Apologies for the ticket
